Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,479 @@
+Parser:
+  Title: Network Session ASIM parser for SonicWall firewalls
+  Version: '0.1.0'
+  LastUpdated: Jan 26, 2024
+Product:
+  Name: SonicWall
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.6'
+References:
+  - Title: ASIM Network Session Schema
+    Link: https://aka.ms/ASimNetworkSessionDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+  - Title: SonicWall SonicOS/X 7.0.1 Log Events Reference Guide [PDF]
+    Link: https://www.sonicwall.com/techdocs/pdf/SonicOS-X_7.0.1_LogEvents_ReferenceGuide.pdf
+Description: |
+  This ASIM parser supports normalizing SonicWall SonicOS ArcSight-formatted Syslog data ingested by Microsoft Sentinel into CommonSecurityLogs to the ASIM Network Session normalized schema.
+ParserName: ASimNetworkSessionSonicWallFirewall
+EquivalentBuiltInParser: _ASim_NetworkSession_SonicWallFirewall
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let Actions=datatable(fw_action:string,DvcAction:string)
+  [ "reset client","Reset Source"
+  , "reset server","Reset Destination"
+  , "reset both", "Reset" 
+  , "allow","Allow"
+  , "\"forward\"","Allow"
+  , "\"mgmt\"","Other"
+  , "\"NA\"","Other"
+  , "deny","Deny"
+  , "\"drop\"", "Drop"
+  , "drop ICMP", "Drop ICMP"];
+  let Parser=(disabled:bool=false){
+  CommonSecurityLog
+  | where not(disabled)
+  | where DeviceVendor == "SonicWall"
+  | where DeviceEventClassID !in (14, 97, 1382)

```

Comment:
@jaimeesc 
I think we should list only Network specific EventIDs here. This condition will include non-network specific events as well. E.g. 744 to 759 (seems like Authentication events), also few are 'System' event. I would suggest going through this documentation and create an array of just network related events and using that as a filter.

Reply:
For this item, rather than filtering on specific IDs only, I am proposing a mix of ID exclusions and event category inclusions. The reason for this is because there are many event IDs that we would want to include that are spread throughout a number of event categories. Additionally, there are some IDs within desired categories that we wouldn't want in the results. So I pre-filter out those IDs and then include only specific event categories with network-specific events. What I'm proposing reduces the maintenance required while still filtering down to network events. I'm still testing this and will adjust as needed.